src/LogFileReader.cpp

```diff
@@ -67,19 +71,35 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    QStringList lines = line.split(COMMA);
+    QTime time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool ok;
+    if (time.isValid() && lines.count() == 3)
+    {
+        batteryData.time = time;
+    }
+    else
+    {
+        return false;
+    }
 
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
-    return true;
+
+    if((lines[1].toDouble(&ok) == ok) || (lines[2].toDouble(&ok) == ok))
```

Comment:
lines[1].toDouble(&ok) would convert that variable into a double, and if the operation fail (not a number) it would convert that variable into 0 with a boolean value of ok. That's why in that conditional sentence, i'm setting if there are booleans values which mean the convert failed and should be return false